inc/toolbox.class.php

```diff
@@ -941,111 +927,87 @@ static function commonCheckForUseGLPI() {
-      }
-      echo "</tr>";
-
-      // Test for curl extension.
-      echo "<tr class='tab_bg_1'><td class='left b'>".__('Curl extension test')."</td>";
-
-      if (!extension_loaded('curl')) {
-         echo "<td><img src='".$CFG_GLPI['root_doc']."/pics/ko_min.png'>".
-                     __('Curl extension of your parser PHP is not installed')."></td>";
-         $error = 2;
+      $needed_extentions = [
+         'mysqli'   => [
+            'title'     => __('MySQL Improved extension test'),
+            'success'   => __s('Ok - the MySQLi class exist - Perfect!'),
+            'error'     => __('You must install the MySQL Improved extension for PHP.')
+         ],
+         'ctype'    => [
+            'title'     => __('Test ctype functions'),
+            'success'   => __s('The functionality is found - Perfect!'),
+            'error'     => __("GLPI can't work correctly without the ctype functions"),
+            'functions' => ['ctype_digit']
+         ],
+         'fileinfo' => [
+            'title'     => __('Fileinfo extension test'),
+            'success'   => __s('The functionality is found - Perfect!'),
+            'error'     => __("Fileinfo extension of your parser PHP is not installed"),
+            'class'     => 'finfo'
+         ],
+         'json'     => [
+            'title'     => __('Test json functions'),
+            'success'   => __s('The functionality is found - Perfect!'),
+            'error'     => __("GLPI can't work correctly without the json_encode and json_decode functions"),
+            'functions' => [
+               'json_encode',
+               'json_decode'
+            ]
+         ],
+         'mbstring' => [
+            'title'     => __('Mbstring extension test'),
+            'success'   => __s('The functionality is found - Perfect!'),
+            'error'     => __('Mbstring extension of your parser PHP is not installed')
+         ],
+         'gd'       => [
+            'title'     => __('GD extension test'),
+            'success'   => __s('The functionality is found - Perfect!'),
+            'error'     => __('GD extension of your parser PHP is not installed')
+         ],
+         'zlib'     => [
+            'title'     => __('Zlib extension test'),
+            'success'   => __s('The functionality is found - Perfect!'),
+            'error'     => __('Zlib extension of your parser PHP is not installed')
+         ],
+         'curl'      => [
+            'title'     => __('Curl extension test'),
+            'success'   => __s('The functionality is found - Perfect!'),
+            'error'     => __('Curl extension of your parser PHP is not installed')
+         ]
+      ];
+
+      $loaded_extensions = [];
+      //check for PHP extensions
+      foreach ($needed_extentions as $ext => $params) {
+         $success = true;
+
+         if (isset($params['functions'])) {
+            foreach ($params['functions'] as $function) {
```

Comment:
I'm not sure testing various functions is really useful, a single one should be enough.